google-cloud-spanner/src/main/java/com/google/cloud/spanner/SpannerCloudMonitoringExporterUtils.java

```diff
@@ -0,0 +1,211 @@
+import com.google.api.Distribution.BucketOptions;
+import com.google.api.Distribution.BucketOptions.Explicit;
+import com.google.api.Metric;
+import com.google.api.MetricDescriptor.MetricKind;
+import com.google.api.MetricDescriptor.ValueType;
+import com.google.api.MonitoredResource;
+import com.google.monitoring.v3.Point;
+import com.google.monitoring.v3.TimeInterval;
+import com.google.monitoring.v3.TimeSeries;
+import com.google.monitoring.v3.TypedValue;
+import com.google.protobuf.util.Timestamps;
+import io.opentelemetry.api.common.AttributeKey;
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.sdk.metrics.data.AggregationTemporality;
+import io.opentelemetry.sdk.metrics.data.DoublePointData;
+import io.opentelemetry.sdk.metrics.data.HistogramData;
+import io.opentelemetry.sdk.metrics.data.HistogramPointData;
+import io.opentelemetry.sdk.metrics.data.LongPointData;
+import io.opentelemetry.sdk.metrics.data.MetricData;
+import io.opentelemetry.sdk.metrics.data.MetricDataType;
+import io.opentelemetry.sdk.metrics.data.PointData;
+import io.opentelemetry.sdk.metrics.data.SumData;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+class SpannerCloudMonitoringExporterUtils {
+
+  private static final Logger logger =
+      Logger.getLogger(SpannerCloudMonitoringExporterUtils.class.getName());
+
+  private SpannerCloudMonitoringExporterUtils() {}
+
+  static String getProjectId(PointData pointData) {
+    return pointData.getAttributes().get(PROJECT_ID_KEY);
+  }
+
+  static List<TimeSeries> convertToSpannerTimeSeries(List<MetricData> collection) {
+    List<TimeSeries> allTimeSeries = new ArrayList<>();
+
+    for (MetricData metricData : collection) {
+      // Get common metrics data from GAX library
+      if (!metricData.getInstrumentationScopeInfo().getName().equals(GAX_METER_NAME)) {
+        // Filter out metric data for instruments that are not part of the spanner metrics list
+        continue;
+      }
+      metricData.getData().getPoints().stream()
+          .map(pointData -> convertPointToSpannerTimeSeries(metricData, pointData))
+          .forEach(allTimeSeries::add);
+    }
+
+    return allTimeSeries;
+  }
+
+  private static TimeSeries convertPointToSpannerTimeSeries(
+      MetricData metricData, PointData pointData) {
+    TimeSeries.Builder builder =
+        TimeSeries.newBuilder()
+            .setMetricKind(convertMetricKind(metricData))
+            .setValueType(convertValueType(metricData.getType()));
+    Metric.Builder metricBuilder = Metric.newBuilder().setType(metricData.getName());
+
+    Attributes attributes = pointData.getAttributes();
+    MonitoredResource.Builder monitoredResourceBuilder =
+        MonitoredResource.newBuilder().setType(SPANNER_RESOURCE_TYPE);
+
+    for (AttributeKey<?> key : attributes.asMap().keySet()) {
+      if (SPANNER_PROMOTED_RESOURCE_LABELS.contains(key)) {
```

Comment:
It's interesting all the MonitoredResource labels are coming from metric level attributes instead of OTel Resource attributes. None of these could be set in the OTel resource?

Reply:
@aabmass It can be set, but does it really make a difference if these Monitored resource labels comes from metrics attributes or OTel resource attributes ?